lib/sqlalchemy/orm/decl_api.py

```diff
@@ -1250,9 +1250,16 @@ def _resolve_type(
             python_type_type = flatten_newtype(python_type)
             search = ((python_type, python_type_type),)
         elif is_pep695(python_type):
-            python_type_type = python_type.__value__
-            flattened = None
-            search = ((python_type, python_type_type),)
+            if is_literal(python_type.__value__):
```

Comment:
copying the suggestion from the issue:
we can probably replace this elif with the following at the top of this method
```py
if is_pep695(python_type):
  return self._resolve_type(python_type.__value__)

Reply:
Not sure I misunderstood, I removed this entire `elif` and added the two lines you suggested at the top, that causes an existing test to fail.

```
FAILED test/orm/declarative/test_typed_mapping.py::MappedColumnTest::test_pep695_typealias_as_typemap_keys - AttributeError: 'types.UnionType' object has no attribute '__mro__'. Did you mean: '__or__'?
```

Reply:
indeed something more elaborate is needed. will think of something

Reply:
I've updated on gerrit https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5260